benchmarks/generate_benchmark_params.py

```diff
@@ -0,0 +1,521 @@
+"""Benchmarking parameters generation file."""
+
+import argparse
+from itertools import chain, product
+
+import pandas as pd
+
+parser = argparse.ArgumentParser(
+    description="Generate parameters for which benchmark is run"
+)
+
+parser.add_argument(
+    "-m",
+    "--manifold",
+    type=str,
+    default="all",
+    help="Manifold for which benchmark is run. 'all' denotes all manifolds present.",
+)
+parser.add_argument(
+    "-n",
+    "--n_samples",
+    type=int,
+    default=10,
+    help="Number of samples for which benchmark is run",
+)
+args = parser.parse_args()
+
+
+def spd_manifold_params(n_samples):
+    """Generate spd manifold benchmarking parameters.
+
+    Parameters
+    ----------
+        n_samples : int
+            Number of samples to be used.
+
+    Returns
+    -------
+        _:list.
+        List of params.
+    """
+    manifold = "SPDMatrices"
+    manifold_args = [(2,), (5,), (10,)]
+    module = "geomstats.geometry.spd_matrices"
+
+    def affine_metric_params():
+        params = []
+        metric = "SPDMetricAffine"
+        power_args = [-0.5, 1, 0.5]
+        metric_args = list(product([item for item, in manifold_args], power_args))
+        manifold_args_re = [
+            item for item in manifold_args for i in range(len(power_args))
+        ]
+        kwargs = {}
+        common = (manifold, module, metric, n_samples, kwargs)
+        for manifold_arg, metric_arg in zip(manifold_args_re, metric_args):
+            params += [common + (manifold_arg, metric_arg)]
+        return params
+
+    def bures_wasserstein_metric_params():
+        params = []
+        metric = "SPDMetricBuresWasserstein"
+        metric_args = manifold_args
+        kwargs = {}
+        common = (manifold, module, metric, n_samples, kwargs)
+        for manifold_arg, metric_arg in zip(manifold_args, metric_args):
+            params += [common + (manifold_arg, metric_arg)]
+        return params
+
+    def euclidean_metric_params():
+        params = []
+        metric = "SPDMetricEuclidean"
+        power_args = [-0.5, 1, 0.5]
+        metric_args = list(product([item for item, in manifold_args], power_args))
+        manifold_args_re = [
+            item for item in manifold_args for i in range(len(power_args))
+        ]
+        kwargs = {}
+        common = (manifold, module, metric, n_samples, kwargs)
+        for manifold_arg, metric_arg in zip(manifold_args_re, metric_args):
+            params += [common + (manifold_arg, metric_arg)]
+        return params
+
+    def log_euclidean_metric_params():
+        params = []
+        metric = "SPDMetricLogEuclidean"
+        metric_args = manifold_args
+        kwargs = {}
+        common = (manifold, module, metric, n_samples, kwargs)
+        for manifold_arg, metric_arg in zip(manifold_args, metric_args):
+            params += [common + (manifold_arg, metric_arg)]
+        return params
```

Comment:
The lines:
```
        kwargs = {}
        common = (manifold, module, metric, n_samples, kwargs)
        for manifold_arg, metric_arg in zip(manifold_args, metric_args):
            params += [common + (manifold_arg, metric_arg)]
        return params
```
seems to be repeated for the params of each metric. 

Should they be factorized? (maybe not, maybe it is more readable/editable this way?)

Reply:
yup.. they could be factorized. I didn't want to over simplify things without knowing completely. Let's say if some metric requires different  processing from information geometry.